xDrip/Components/Charts/GlucoseHistoryView.swift

```diff
@@ -127,9 +156,16 @@ final class GlucoseHistoryView: UIView {
     }
     
     /// Should be sorted by date ascending
-    func setup(with entries: [GlucoseChartGlucoseEntry], unit: String) {
+    func setup(
+        with entries: [GlucoseChartGlucoseEntry],
+        basalEntries: [BasalChartEntry],
+        basalRates: [BasalRate],
```

Comment:
You shouldn't pass `BasalRate` to a view and do calculations there, because this is business logic. Do all the calculations in the interactor (or presenter) prior to passing it to the view.